endpoint.go

```diff
@@ -417,36 +297,30 @@ func (ep *endpoint) Join(containerID string, options ...EndpointOption) error {
 		}
 	}()
 
-	err = ep.buildHostsFiles()
+	err = sb.buildHostsFiles(ep)
 	if err != nil {
 		return err
 	}
 
-	err = ep.updateParentHosts()
+	err = sb.updateParentHosts(ep)
 	if err != nil {
 		return err
 	}
 
-	err = ep.setupDNS()
+	err = sb.setupDNS(ep)
```

Comment:
The issue here is that setupDNS needs to know if ipv6 is enabled on the network, and at time of sandbox creation it is not known to which network the sandbox will attach.